### PR TITLE
fix: 🤔 (unchecked) checkbox in sd-select does not keep size while resizing sd-popup

### LINKED ISSUE #1058: fix: 🤔 (unchecked) checkbox in `sd-select` does not keep size while resizing `sd-popup`
## Current behavior
When resizing the window (and the `sd-popup` inside the `sd-select`) the unchecked checkbox reduced the with as well. 
## Expected behavior
The unchecked checkbox keeps the size.

## Steps to reproduce
See attached video in https://jira.apps.union-investment.de/browse/PSOLIDDS-29
## DoR
- [x] Item has business value
- [x] Item has been estimated by the team
- [x] Item is clear and well-defined
- [x] Item dependencies have been identified

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch

### PR DESCRIPTION
**Pull request by Vahid1919**
## Description:

Closes  #1058. 

## Definition of Reviewable:

- [x] relevant tickets are linked

### CODE REVIEW

**Review by smfonseca — APPROVED** [review]

File: packages/components/src/components/option/option.ts [hunk 1]
```diff
@@ -157,7 +157,7 @@ export default class SdOption extends SolidElement {
               id="control"
               part="control ${this.selected ? ' control--checked' : 'control--unchecked'}"
               class=${cx(
-                'relative flex flex-initial items-center justify-center border rounded-sm h-4 w-4 mr-2',
+                'relative flex flex-shrink-0 items-center justify-center border rounded-sm h-4 w-4 mr-2',
```
Comment (karlbaumhauer) [inline packages/components/src/components/option/option.ts:160]: I think it needs to be fixed in the checkbox itself... testing it there, still shows the bug:


Uploading Screen Recording 2024-09-17 at 18.50.49.mov…

**Review by karlbaumhauer — CHANGES_REQUESTED** [review]

**Comment by coraliefeil** [comment]
@MartaPintoTeixeira I’ll leave the review to you

File: packages/components/src/components/option/option.ts (hunk 1 above)
Comment (Vahid1919) [reply packages/components/src/components/option/option.ts:160]: @karlbaumhauer From what I see, you are manually adding a width element to the checkbox, the size changing there should be expected. I think a better test would be to change the width of the window with your cursor, or adding even more text to the slot. In both cases, the checkbox alone retains it's size. 
image

**Review by yoezlem — APPROVED** [review]

File: packages/components/src/components/option/option.ts (hunk 1 above)
Comment (Vahid1919) [reply packages/components/src/components/option/option.ts:160]: Updated.

**Review by karlbaumhauer — APPROVED** [review]

**Comment by karlbaumhauer** [comment]
@Vahid1919 approved it but now I forgot to mention the test. Pls add a screenshot test for this (maybe something like the original issue eg. a long lable).

**Comment by MartaPintoTeixeira** [comment]
Approved thanks.

**Pull request closed without merging**